scripts/Utils.py

```diff
@@ -110,18 +113,40 @@ def UnzipFile(filepath, deleteZipFile=True):
             except ZeroDivisionError:
                 done = 50
                 percentage = 100
-            elapsedTime = time.time() - startTime
+            elapsedTime = time.time() - startTime            
+            estimatedTime = 0
             try:
                 avgKBPerSecond = (extractedContentSize / 1024) / elapsedTime
+                estimatedTime = ( (zipFileContentSize - extractedContentSize) / (avgKBPerSecond*1024) )
             except ZeroDivisionError:
                 avgKBPerSecond = 0.0
+                estimatedTime = 0
+            estimatedTimeString = FormatTime(estimatedTime)
             avgSpeedString = '{:.2f} KB/s'.format(avgKBPerSecond)
```

Comment:
Same comment as on line 82

For flow of the program, quite a small detail, but I would suggest putting the speed first and then the time, as we first calculated the speed as well...
```suggestion
            avgSpeedString = '{:.2f} KB/s'.format(avgKBPerSecond)
            estimatedTimeString = FormatTime(estimatedTime)
```